csrc/host_ir/executor.cpp

```diff
@@ -46,16 +45,9 @@ HostIrExecutor::HostIrExecutor(
 bool HostIrExecutor::supported(Fusion* fusion) {
   FUSER_PERF_SCOPE("HostIrExecutor::supported");
   std::vector<Expr*> exprs = fusion->exprs();
-  if (std::any_of(exprs.begin(), exprs.end(), [](Expr* e) {
-        return isResharding(e) && HostIrLower::canLower(e);
-      })) {
+  if (std::any_of(exprs.begin(), exprs.end(), isResharding)) {
     NVF_ERROR(
-        std::all_of(
-            exprs.begin(),
-            exprs.end(),
-            [](Expr* e) {
-              return isResharding(e) && HostIrLower::canLower(e);
-            }),
+        std::all_of(exprs.begin(), exprs.end(), isResharding),
```

Comment:
I recall isResharding being an expensive function after switching from IdModel so we can also use `count_if` here instead,
Since we may only be dealing with one expression here (due to how communication scheduler works) -- it may not be as useful at the moment.

Reply:
The real problem is that https://github.com/NVIDIA/Fuser/blob/1b4ce020d354c82178a1a39491803e01aca028b5/csrc/runtime/executor_dispatch.cpp#L27 is called before other Executor::supported. This means a potentially large non-communication segment will trigger isResharding for every Expr there due to `std::any_of` and **regardless** of `std::all_of`. 

I'll verify this assumption in a separate PR. I think it's a matter of calling https://github.com/NVIDIA/Fuser/blob/1b4ce020d354c82178a1a39491803e01aca028b5/csrc/runtime/executor_dispatch.cpp#L27 the last.